buku

```diff
@@ -5668,7 +5685,7 @@ def parse_range(tokens: Optional[str | Sequence[str] | Set[str]],  # Optional[st
 
 
 # main starts here
-def main():
+def main(argv=sys.argv[1:], *, program_name=os.path.basename(sys.argv[0])):
```

Comment:
This does not affect current usage, but allows to invoke `main()` without relying on `sys.argv` (i.e. for testing)

Reply:
Note that slicing is necessary for that purpose (since `ArgumentParser.parse_args()` ignores the 1st value in the list _only_ if the list is `sys.argv`)